string namespace_base
bool availableOnly
---
bool ack
